### PR TITLE
CVE-2021-22942 actionpack possible open redirect

### PR DESCRIPTION
**Pull request by lukepearcewp**

### CODE REVIEW

File: gems/actionpack/CVE-2021-22942.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  * 6-0-host-authorzation-open-redirect.patch - Patch for 6.0 series
+  * 6-1-host-authorzation-open-redirect.patch - Patch for 6.1 series
+
+  Please note that only the 6.1.Z, 6.0.Z, and 5.2.Z series are supported at present. Users of earlier unsupported releases are advised to upgrade as soon as possible as we cannot guarantee the continued availability of security fixes for unsupported releases.
+
+unaffected_versions:
+- < 6.0.0
+patched_versions:
+- '>= 6.0.4.1'
```
Comment (reedloden) [inline gems/actionpack/CVE-2021-22942.yml:66]: This should be:

'~> 6.0.4, >= 6.0.4.1'

File: gems/actionpack/CVE-2021-22942.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  Impacted applications will have allowed hosts with a leading dot. For example, configuration files that look like this:
+
+  ```ruby
+  config.hosts <<  '.EXAMPLE.com'
+  ```
+
+  When an allowed host contains a leading dot, a specially crafted Host header can be used to redirect to a malicious website.
+
+  This vulnerability is similar to CVE-2021-22881, but CVE-2021-22881 did not take in to account domain name case sensitivity.
+  Releases
```
Comment (reedloden) [inline gems/actionpack/CVE-2021-22942.yml:27]: Needs newline before it and dotted lines beneath it like the other section headers.

File: gems/actionpack/CVE-2021-22942.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
+---
+gem: actionpack
+framework: rails
+cve: 2021-22942
+url: https://groups.google.com/g/rubyonrails-security/c/wB5tRn7h36c
+title: Possible Open Redirect in Host Authorization Middleware
+date: 2021-08-19
+description: |
+  There is a possible open redirect vulnerability in the Host Authorization middleware in Action Pack. This vulnerability has been assigned the CVE identifier CVE-2021-22942.
```
Comment (reedloden) [inline gems/actionpack/CVE-2021-22942.yml:9]: I'm doing this from mobile, so hard for me to count, but several of these lines seem pretty long. We prefer wrapping somewhere in the 80-100 character range (see other files). Can you check and see if any of these are too long and need to be wrapped?

File: gems/actionpack/CVE-2021-22942.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        def authorized?(request)
+          origin_host = request.get_header("HTTP_HOST")&.slice(VALID_ORIGIN_HOST, 1) || ""
+          forwarded_host = request.x_forwarded_host&.slice(VALID_FORWARDED_HOST, 1) || ""
+          @permissions.allows?(origin_host) && (forwarded_host.blank? || @permissions.allows?(forwarded_host))
+        end
+    end
+  end
+  ```
+
+  Patches
```
Comment (reedloden) [inline gems/actionpack/CVE-2021-22942.yml:53]: Please remove this entire section. We don't provide the patches, so this is a bit confusing.

File: gems/actionpack/CVE-2021-22942.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  -------
+
+  To aid users who aren’t able to upgrade immediately we have provided patches for the two supported release series. They are in git-am format and consist of a single changeset.
+
+  * 6-0-host-authorzation-open-redirect.patch - Patch for 6.0 series
+  * 6-1-host-authorzation-open-redirect.patch - Patch for 6.1 series
+
+  Please note that only the 6.1.Z, 6.0.Z, and 5.2.Z series are supported at present. Users of earlier unsupported releases are advised to upgrade as soon as possible as we cannot guarantee the continued availability of security fixes for unsupported releases.
+
+unaffected_versions:
```
Comment (reedloden) [inline gems/actionpack/CVE-2021-22942.yml:63]: Before this, can we add `cvss_v3` of 7.6, as per https://access.redhat.com/security/cve/cve-2021-22942

**Review by reedloden — CHANGES_REQUESTED** [review]
Thanks so much for submitting this! A few minor changes needed before we can merge this. Really appreciate your work on this!

File: gems/actionpack/CVE-2021-22942.yml (hunk 5 above)
Comment (lukepearcewp) [reply gems/actionpack/CVE-2021-22942.yml:68]: Ah thanks, I wasn't sure where to find that, my quick google search didn't come up with it.

File: gems/actionpack/CVE-2021-22942.yml (hunk 2 above)
Comment (lukepearcewp) [reply gems/actionpack/CVE-2021-22942.yml:37]: Fixed

File: gems/actionpack/CVE-2021-22942.yml (hunk 1 above)
Comment (lukepearcewp) [reply gems/actionpack/CVE-2021-22942.yml]: Done

File: gems/actionpack/CVE-2021-22942.yml (hunk 3 above)
Comment (lukepearcewp) [reply gems/actionpack/CVE-2021-22942.yml]: Serves me right for doing a ninja edit in the github editor :)

File: gems/actionpack/CVE-2021-22942.yml (hunk 4 above)
Comment (lukepearcewp) [reply gems/actionpack/CVE-2021-22942.yml]: done

**Review by reedloden — APPROVED** [review]
Thanks for the quick turnaround! Doing this review from mobile, so hoping I don't miss anything. Fingers crossed.

**Comment by lukepearcewp** [comment]
Nice one, tested on a local project all works correctly for me.

**Pull request merged**